meter/meter.go

```diff
@@ -74,7 +74,21 @@ func NewConfigurableFromConfig(other map[string]interface{}) (api.Meter, error)
 		}
 	}
 
-	res := m.Decorate(totalEnergyG, currentsG, voltagesG, powersG, batterySocG, cc.capacity.Decorator())
+	// decorate capacity
+	var capacityStruct capacity
+	if cc.Capacity != nil {
+		capacityG, floatGetterErr := provider.NewFloatGetterFromConfig(*cc.Capacity)
+		if floatGetterErr != nil {
+			return nil, fmt.Errorf("error parsing capacity structure: %w", floatGetterErr)
+		}
+		readCapacity, capacityErr := capacityG()
+		if capacityErr != nil {
+			return nil, fmt.Errorf("error retrieving capacity: %w", capacityErr)
+		}
+		capacityStruct.Capacity = readCapacity
```

Comment:
> Do you mean removing the cc struct?

This struct should not be needed here. Only purpose was adding the battery api, that can be done simpler.

Reply:
There are still some other implementations relying on this so I don´t see an easy way to get rid of it:
![image](https://github.com/evcc-io/evcc/assets/45375059/3aa19726-7438-4ea6-9148-4523bf6fdc6a)
We can simplify to
```go
	var capacityG func() float64
	if cc.Capacity != nil {
		capacityErrG, floatGetterErr := provider.NewFloatGetterFromConfig(*cc.Capacity)
		if floatGetterErr != nil {
			return nil, fmt.Errorf("error parsing capacity structure: %w", floatGetterErr)
		}
		readCapacity, capacityErr := capacityErrG()
		if capacityErr != nil {
			return nil, fmt.Errorf("error retrieving capacity: %w", capacityErr)
		}
		capacityG = func() float64 {
			return readCapacity
		}
	}

	res := m.Decorate(totalEnergyG, currentsG, voltagesG, powersG, batterySocG, capacityG)
```
but is that better than using the struct?

Reply:
Yes 👍🏻